src/api/HttpApi.ts

```diff
@@ -0,0 +1,237 @@
+import querystring from 'querystring';
+import DataLoader from 'dataloader';
+import {
+  Connection,
+  connectionFromArray,
+  forwardConnectionArgs,
+} from 'graphql-relay';
+import invariant from 'invariant';
+import chunk from 'lodash/chunk';
+import flatten from 'lodash/flatten';
+import omit from 'lodash/omit';
+import pick from 'lodash/pick';
+
+import urlJoin from '../utils/urlJoin';
+import { HttpMethod } from './fetch';
+
+const PAGINATION_ARG_KEYS = Object.keys(forwardConnectionArgs);
+
+export type Args = Record<string, NodeJS.PoorMansUnknown>;
+export type Data = unknown | null | undefined;
+
+export type PaginationResult<T> = Connection<T> & {
+  meta: {};
+};
+
+export type PaginatedApiResult<T> = Array<T> & {
+  meta: {
+    cursors?: any;
+    hasNextPage?: boolean;
+  };
+};
+export type HttpApiOptions = {
+  apiBase: string;
+  origin: string;
+  externalOrigin: string;
+};
+
+export default class HttpApi {
+  _origin: string;
+
+  _apiBase: string;
+
+  _externalOrigin: string;
+
+  _loader: DataLoader<any, any>;
+
+  qs = querystring;
```

Comment:
Removed the `QueryString` type as we can just get it from `@types/querystring`

Reply:
We don't want the interface to be exactly the query string package necessarily, this should define the minimal interface for overriding